docs/proposals/20220126_k1-for-devs.md

```diff
@@ -0,0 +1,76 @@
+# KubeOne for Developers
```

Comment:
This proposal is missing some key sections:

* Values and alternatives -- the key question I'd like to see answered is why would someone want to use this over GitPod and GitHub Codespaces? Those two products are also cheaper than a Kubernetes cluster created by KubeOne on any cloud provider, so there's that too.
* Ownership -- who is going to own and maintain the project?
* Location -- where is this addon going to live? This depends on the ownership question as well. If this is to be maintained by a dedicated team then it should live in another repo.

In my opinion, everyone should be able to use KubeOne as a building block. In that aspect, I'd like this proposal to focus on what we can do to make implementing this possible. I see two key aspects:

* Improving usability and the infrastructure management experience
* Improving the addons mechanism

If that's the case, I'd like to see what we can do to improve those two aspects.

Reply:
**Values and alternatives** Yes, I think we should spent some time looking for alternatives on the market before starting implementation. Concerning GitPod and CodeSpaces: I am not sure if those are usable on private clusters. Those tools are running on some GitHub infra and not in a dedicated cluster. IMO the idea is to add functionality to clusters created via KubeOne. Which you may can do with GitPod and CodeSpaces via connecting them to your Kubernetes Cluster but this will not be doable all the time (private clusters) and also has some security implications.

**Ownership & Location** Good question, we should discuss this, maybe @scheeles has some input here